schemachanger: add & adopt data elements

postamar opened the pull request:
This commit attaches db/table/index GC jobs to elements representing
their data. This allows leveraging the rules system to create a minimum
of necessary GC jobs in the same non-revertible mutation stage.

Fixes #82169.

Release note: None

Commit:
schemachanger: add & adopt data elements

This commit attaches db/table/index GC jobs to elements representing
their data. This allows leveraging the rules system to create a minimum
of necessary GC jobs in the same non-revertible mutation stage.

Fixes #82169.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/89029)
<!-- Reviewable:end -->

Commit:
scpb: remove IsRelationBeingDropped fields

These have been rendered obsolete by the addition of not-join clauses in
the rules.

Fixes #89002.

Release note: None

postamar commented:
Postponing this change to 23.1.
